src/main/webapp/js/user-page-loader.js

```diff
@@ -228,6 +242,7 @@ function buildUI() {
   setPageTitle();
   showMessageFormIfLoggedIn();
   fetchMessages();
+  fetchGallery();
```

Comment:
I think this approach is fine (and it works with what I tested). But just something to think about: there's a trade-off here between memory and speed. Right now we're storing both the "posts" and the "gallery" tabs in memory, so switching between them is fast. We could also go the other way and only store one of them in memory at a time, and do the fetch when the user switches tabs instead. That would use ~50% as much memory, but it comes at the cost of taking longer to switch tabs.

(No action required here, just food for thought.)

Reply:
Yeah, I was kind of thinking about that as I was originally writing it. The fetches aren't actually that slow and the gallery will probably be viewed less often than the user page so maybe it would be better to reduce the memory consumption and just fetch when needed.